### PR TITLE
feat: #1303 and fix #1294 1307, new modal style

### LINKED ISSUE #1294: [Bug] Mermaid提示Syntax error in text
**Describe the bug**
无法正常渲染mermaid。经过确认应该是mermaid本身的一个兼容问题，流程图中不能出现中文标点，会被识别为unrecognized text，无法正常渲染。因而可能需要向上游提交请求。

**To Reproduce**
```
graph TD;
  A[开始]-->B(if 吃饭了吗?)
  B-- 是 -->C(好的，继续工作)
  B-- 否 -->D{饿了吗?}
  D-- 是 -->E(那去吃吧)
  D-- 否 -->C(好的，继续工作)
  C-->F[结束]
  E-->F
```
上述的中文逗号换成英文逗号可以正常渲染，否则会出现错误。

**Screenshots**
image

### PR DESCRIPTION
**Pull request by Yidadaa**

### CODE REVIEW

File: next.config.mjs [hunk 1]
```diff
@@ -5,7 +5,12 @@ const nextConfig = {
     appDir: true,
   },
   async rewrites() {
-    const ret = [];
+    const ret = [
+      {
+        source: "/api/proxy/:path*",
+        destination: "https://api.openai.com/:path*",
```
Comment (l2dy) [inline next.config.mjs:11]: Is this open proxy necessary for the feature?

Comment (Yidadaa) [reply next.config.mjs:11]: I need this proxy to use other chatgpt-based tools.

Comment (l2dy) [reply next.config.mjs:11]: @Yidadaa An open proxy could be abused by bad actors. Please consider securing this endpoint for deployments with `CODE` configured.

Comment (Yidadaa) [reply next.config.mjs:11]: I am trying to make it, this is a trade-off solution.

**Pull request merged**